### DISCUSSION TITLE
What is the reason to rename `--mdc-*` to `--md-*`

### DISCUSSION THREAD

**Discussion by scherler** [material-components/material-web#5167] [General]
Doing the update of our development from `mwc` to `md` would save quite some time if I did not have to rename all our CSS variables, which leads me to the question of why the decision was taken to rename the CSS variable. I do understand the tags but the variables could have been the same.

**Comment by asyncliz** [comment]
The variable names would be completely different due to the naming schema that was designed for Material 3 tokens.

Since none of them would be compatible, we used a different prefix to signify the breaking change.

A smoother upgrade from `<mwc-*>` to `<md-*>` is on our radar though. We have many folks at Google with the same problem! I'm hoping to have it as a goal for 2024.

**Reply by scherler** [reply]
Thank you very much for your response and yes makes sense with the variables. Regarding the update route, some of the new components do not support fundamental methods e.g. `md-menu` does not support `@action` anymore, it may be a good idea to drop a line in the docu on how to migrate your `<mwc-menu @action ...` code, I am using the menu as an example since I am currently working on it :wink: but I would await this on any component.

**Reply by asyncliz** [reply]
I would love to have good documentation for migration support!

The tl;dr of our plan is to create something like `@material/web/migrations/m2/mwc-button` that would call `<md-*>` implementations under the hood, style itself to look like M2, and try to map as many 1:1 APIs like methods, events, and custom properties.

Later on, the same concept can be used for breaking changes that require migrations, such as a re-design of a component. Slider's on the chopping block for a re-design soon for example.

There are _very_ few engineers on MWC at the moment though, so if this sounds like a fun project to explore we'd welcome contributions! You can also chat with us on [lit.dev/discord](https://lit.dev/discord) in the #material channel :)
